File: output/CodeGen/avatar/Java/Python/codeforces_617_B.py
def main(args):
    n = input()
    arr = []
    for i in range(n):
        if input() == 1:
            arr.append(i)
    if len(arr) == 0:
        print(0)
    else:
        result = 1
        for i in range(1, len(arr)):
            result *= arr[i] - arr[i-1]
        print(result)

File: output/ChatGPT/avatar/Java/C++/codeforces_617_B.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> arr;
    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        if (num == 1) {
            arr.push_back(i);
        }
    }

    if (arr.empty()) {
        cout << 0 << endl;
    } else {
        long long result = 1;
        for (int i = 1; i < arr.size(); i++) {
            result *= arr[i] - arr[i - 1];
        }
        cout << result << endl;
    }

}